src/explore-education-statistics-admin/src/pages/release/pre-release/components/__tests__/PreReleaseUserAccessForm.test.tsx

```diff
@@ -206,58 +253,248 @@ describe('PreReleaseUserAccessForm', () => {
 
       await waitFor(() => {
         expect(
-          screen.getByLabelText('Invite new user by email'),
+          screen.getByLabelText('Invite new users by email'),
         ).toBeInTheDocument();
       });
 
-      userEvent.click(screen.getByRole('button', { name: 'Invite new user' }));
+      userEvent.click(screen.getByRole('button', { name: 'Invite new users' }));
 
       await waitFor(() => {
         expect(
-          screen.getByText('Enter an email address', {
-            selector: '#preReleaseUserAccessForm-email-error',
+          screen.getByText('Please enter 1 or more email addresses', {
+            selector: '#preReleaseUserAccessForm-emails-error',
           }),
         ).toBeInTheDocument();
       });
     });
 
-    test('submitting form successfully adds newly invited user to list', async () => {
+    test('submitting the form opens confirmation modal with invite plan', async () => {
+      preReleaseUserService.getUsers.mockResolvedValue(testUsers);
+
+      render(<PreReleaseUserAccessForm releaseId="release-1" />);
+
+      await waitFor(() => {
+        expect(
+          screen.getByLabelText('Invite new users by email'),
+        ).toBeInTheDocument();
+      });
+
+      await userEvent.type(
+        screen.getByLabelText('Invite new users by email'),
+        'test1@education.gov.uk{enter}test2@education.gov.uk{enter}test3@education.gov.uk',
+      );
+
+      preReleaseUserService.getInvitePlan.mockResolvedValue({
+        alreadyAccepted: [
+          'existing.prerelease.user.1@education.gov.uk',
+          'existing.prerelease.user.2@education.gov.uk',
+        ],
+        alreadyInvited: [
+          'invited.prerelease.1@education.gov.uk',
+          'invited.prerelease.2@education.gov.uk',
+        ],
+        invitable: [
+          'test1@education.gov.uk',
+          'test2@education.gov.uk',
+          'test3@education.gov.uk',
+        ],
+      });
+
+      userEvent.click(screen.getByRole('button', { name: 'Invite new users' }));
+
+      await waitFor(() => {
+        expect(preReleaseUserService.getInvitePlan).toHaveBeenCalledWith(
+          'release-1',
+          'test1@education.gov.uk\ntest2@education.gov.uk\ntest3@education.gov.uk',
+        );
+      });
+
+      await waitFor(() => {
+        expect(screen.getByRole('dialog')).toBeInTheDocument();
```

Comment:
Mentioned this to Amy before, but `getByRole` is a little problematic when used within `waitFor`. It's not a quick selector as it has to do a bunch of visibility checks that are really slow (currently).

This means that if the component is big enough, it can cause `waitFor` (this is essentially just a loop that retries the callback every x milliseconds) to get jammed up with callbacks that are still processing. You can often see tests that are written like this can be noticeably slower, or in extreme cases, be unable to finish within the default 10 second time limit!

Consequently, my current advice is to use faster selectors such as `getByText` or `getByLabel` within the `waitFor` and use `getByRole` outside.

In this specific case, we might do something do something like this:

```ts
await waitFor(() => {
  expect(screen.getByText('Confirm pre-release invitations')).toBeInTheDocument();
});

// Now we can use `getByRole`
const modal = within(screen.getByRole('dialog'));

// We might also check that it's a heading within the modal
expect(modal.getByRole('heading', { name: 'Confirm pre-release invitations' }))
  .toBeInTheDocument();
```
Would suggest going through these tests and checking for any other such cases and refactoring them accordingly.